lib/index.js

```diff
@@ -1,11 +1,19 @@
 'use strict';
 
+const RateLimiter = require('./rate-limiter');
+
 exports.register = (server, options, next) => {
 
   server.ext('onPostAuth', (request, reply) => {
-    request.payload.loaded = true;
+    return RateLimiter.limit(request, options.redisClient, options.defaultRate)
+    .then((rate) => {
+      if (rate && rate.current > rate.limit) {
+        return reply(options.overLimitHandler());
```

Comment:
@robinjoseph08 @siddhantdange instead of providing an over limit handler callback, what if you just provided the error type to throw if the rate limit was exceed?